File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        double sum = 0;
        int rem, temp;
        temp = x;
        while(x > 0){
            rem = x%10;
            sum = sum*10+rem;
            x = x/10;
        }

        if(temp == sum){
            return true;
        } else {
            return false;
        }
        return false;
    }
};

File: 9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        int sum = 0;
        int originalVal = x;

        if(originalVal < 0){
            return false;
        }

        while(x != 0){
            int rem = x % 10;
            sum = sum*10 + rem;
            x /= 10;
        }

        if(originalVal == sum){
            return true;
        }
        return false;
    }
}